src/realm-settings/PartialImport.tsx

```diff
@@ -278,107 +285,258 @@ export const PartialImportDialog = (props: PartialImportProps) => {
-                  onSelect={handleRealmSelect}
-                  placeholderText={targetRealm.realm || targetRealm.id}
+                  isOpen={isCollisionSelectOpen}
+                  direction="up"
+                  onToggle={() => {
+                    setIsCollisionSelectOpen(!isCollisionSelectOpen);
+                  }}
+                  onSelect={handleCollisionSelect}
+                  placeholderText={t(collisionOption)}
                 >
-                  {realmSelectOptions()}
+                  {collisionOptions()}
                 </Select>
               </StackItem>
-            )}
-            <StackItem>
-              <Text>{t("chooseResources")}:</Text>
-              <DataList aria-label="Resources to import" isCompact>
-                {targetHasResource("users") &&
-                  resourceDataListItem("users", "users")}
-                {targetHasResource("groups") &&
-                  resourceDataListItem("groups", "groups")}
-                {targetHasResource("clients") &&
-                  resourceDataListItem("clients", "clients")}
-                {targetHasResource("identityProviders") &&
-                  resourceDataListItem(
-                    "identityProviders",
-                    "identity providers"
-                  )}
-                {targetHasRealmRoles() &&
-                  resourceDataListItem("realmRoles", "realm roles")}
-                {targetHasClientRoles() &&
-                  resourceDataListItem("clientRoles", "client roles")}
-              </DataList>
-            </StackItem>
-            <StackItem>
-              <Text>{t("selectIfResourceExists")}:</Text>
-              <Select
-                isOpen={isCollisionSelectOpen}
-                direction="up"
-                onToggle={() => {
-                  setIsCollisionSelectOpen(!isCollisionSelectOpen);
-                }}
-                onSelect={handleCollisionSelect}
-                placeholderText={t(collisionOption)}
-              >
-                {collisionOptions()}
-              </Select>
-            </StackItem>
-          </>
-        )}
-      </Stack>
-    </Modal>
+            </>
+          )}
+        </Stack>
+      </Modal>
+    );
+  };
+
+  const importCompleteMessage = () => {
+    return t("importSuccess", {
+      added: importResponse?.added,
+      overwritten: importResponse?.overwritten,
+      skipped: importResponse?.skipped,
+    });
+  };
+
+  const loader = async (first?: number, max?: number) => {
+    let last = first! + max!;
+    const length = importResponse?.results.length;
+    if (last > length!) last = length!;
+
+    if (importResponse?.results)
+      return importResponse.results.slice(first, last);
+
+    // Empty state can't happen as you will get back a result record
+    // for each record you submit. This is here to make Typescript happy.
+    return [];
+  };
+
+  const ActionLabel = (importRecord: { action: string }) => {
```

Comment:
You can also directly reference the original type here. It's a tad verbose as the type was not extracted in the admin client library.
```suggestion
  const ActionLabel = (importRecord: PartialImportResponse["results"][0]) => {
```

Reply:
I don't understand that one.  Why would I want to do that?

Reply:
It would allow you to use all the fields from this object without creating a duplicate type for it. However it's perhaps not the best option. Consider this one resolved.